src/main/scala/chisel3/util/circt/PlusArgsValue.scala

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.circt
+
+import chisel3._
+import chisel3.experimental.{FlatIO, IntrinsicModule}
+import chisel3.internal.Builder
+
+import circt.Intrinsic
+
+/** Create a module which generates a verilog \$plusargs\$value.  This returns a
+  * value as indicated by the format string and a flag for whether the value
+  * was found.
+  */
+private class PlusArgsValueIntrinsic[T <: Data](gen: T, str: String)
+    extends IntrinsicModule("circt_plusargs_value", Map("FORMAT" -> str)) {
+  val io = FlatIO(new Bundle {
+    val found = Output(Bool())
+    val result = Output(gen)
+  })
+}
+
+object PlusArgsValue {
+
+  /** Creates an intrinsic which calls \$test\$plusargs.
+    *
+    * @example {{{
+    * b := PlusArgsValue(UInt(32.W), "FOO=%d")
+    * b.found
+    * b.value
+    * }}}
+    */
+  def apply[T <: Data](gen: T, str: String) = {
+    Module(if (gen.isSynthesizable) {
+      new PlusArgsValueIntrinsic(chiselTypeOf(gen), str)
+    } else {
+      new PlusArgsValueIntrinsic(gen, str)
+    }).io
+  }
```

Comment:
Can we add this API:

```suggestion
  }
  
  /** Return the value of a plus arg if set, otherwise return the default value.
    */
  def apply[T <: Data](gen: T, str: String, default: T): T = {
    val result = apply(gen, str)
    Mux(result.found, result.value, default)
  }
```